src/System.Security.Principal.Windows/src/Microsoft/Win32/SafeHandles/SafeSecurityHandles.cs

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Security;
+
+namespace Microsoft.Win32.SafeHandles
+{
+    internal abstract class SafeHandleZeroOrMinusOneIsInvalid : SafeHandle
+    {
+        protected SafeHandleZeroOrMinusOneIsInvalid(bool ownsHandle) : base(IntPtr.Zero, ownsHandle)
+        {
+        }
+
+        public override bool IsInvalid
+        {
+            get
+            { return handle == IntPtr.Zero || handle == new IntPtr(-1); }
+        }
+    }
+
+    internal sealed class SafeLsaMemoryHandle : SafeBuffer
+    {
+        private SafeLsaMemoryHandle() : base(true) { }
+
+        // 0 is an Invalid Handle
+        internal SafeLsaMemoryHandle(IntPtr handle) : base(true)
+        {
+            SetHandle(handle);
+        }
+
+        internal static SafeLsaMemoryHandle InvalidHandle
+        {
+            get { return new SafeLsaMemoryHandle(IntPtr.Zero); }
```

Comment:
Should such an invalid handle be cached rather than allocating one each time the property is accessed?